### PR TITLE
BW-1317 Upgrade `googleCloudNioV` to latest

### PR DESCRIPTION
**Pull request by aednichols**
Last time we tried to upgrade this we [ran into a bug](https://github.com/googleapis/java-storage-nio/issues/691), but it [has now been fixed](https://github.com/googleapis/java-storage-nio/pull/719) and we have a [unit test](https://github.com/broadinstitute/cromwell/pull/6491) for the library that verifies the fix.

Besides generally keeping important libraries up to date, there is a chance that recent fixes to requester pays ([Google PR](https://github.com/googleapis/java-storage-nio/pull/858), [GATK PR](https://github.com/broadinstitute/gatk/pull/7730)) implemented by our own very Methods team may solve an issue users are seeing in Terra:
```
User project specified in the request is invalid.
```

[Terra User Liaison Slack link.](https://broadinstitute.slack.com/archives/CBJJ7U293/p1657731546766919)
[DSP Methods Slack link.](https://broadinstitute.slack.com/archives/C1H82GH41/p1647028919917549?thread_ts=1645714642.593619&cid=C1H82GH41)

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -40,9 +40,7 @@ object Dependencies {
   // latest date via: https://github.com/googleapis/google-api-java-client-services/blob/main/clients/google-api-services-cloudkms/v1.metadata.json
   private val googleCloudKmsV = "v1-rev20220104-1.32.1"
   private val googleCloudMonitoringV = "3.2.5"
-  // BW-808 Pinning googleCloudNioV to this tried-and-true old version and quieting Scala Steward.
-  // 0.121.2 is the most recent version currently known to work.
-  private val googleCloudNioV = "0.61.0-alpha" // scala-steward:off
```
Comment (aednichols) [inline project/Dependencies.scala:45]: It always makes me a little uneasy deleting "danger!" comments but we do have a test and the proof of the pudding will be in the eating

**Comment by breilly2** [comment]
Documenting my review process here.

For dependency updates, I like to review the changelog. In this case, we're going from `0.61.0-alpha` to `0.124.8` which is a large jump, but that doesn't tell the whole story.

* This looks like a lot of releases to check. For sure, checking every release manually is not practical; we'll have to rely on their release notes.
* Until `0.120.0`, this library used to be included in a [monorepo-ish repo of Java libraries](https://github.com/googleapis/google-cloud-java) which appears to have had a regular 2-week release cycle. Not every release had changes to the `google-storage-nio` library. In fact, browsing the release notes, I found only a handful that mentioned changes to storage NIO. These all looked very innocent to me.
* After `0.120.0`, the library code moved to its own [repo](https://github.com/googleapis/java-storage-nio). Releases there have been less frequent and more irregularly scheduled, but still largely consist of dependency updates. (It's possible that _those_ dependency updates introduce unexpected behaviors in `java-storage-nio`, but there's only so much we can audit).
* Cromwell was briefly running with `0.123.8` until the bug mentioned here was discovered. Not knowing when that bug was introduced, we rolled all the way back. Now, we are pretty confident that it was introduced in [`0.122.0`](https://github.com/googleapis/java-storage-nio/releases/tag/v0.122.0) and fixed in [`0.123.13`](https://github.com/googleapis/java-storage-nio/releases/tag/v0.123.13).
* Again looking at releases that are not just dependency updates, nearly all of the changes look very innocent to me. In fact, updating to at least [`0.123.23`](https://github.com/googleapis/java-storage-nio/releases/tag/v0.123.23) will give us the benefit of a [fix](https://github.com/googleapis/java-storage-nio/pull/841) to a requester-pays problem that we encountered ourselves.
* There's only one other post-`0.120.0` [change](https://github.com/googleapis/java-storage-nio/pull/774) (in [`0.123.18`](https://github.com/googleapis/java-storage-nio/releases/tag/v0.123.18)) that raises my eyebrows a little. It's _probably_ fine, but there is new usage of `StorageOptionsUtil.getDefaultInstance()` for which I don't know the lifecycle or how else it's used. This is the type of thing that I'd watch out for in terms of thread safety, which is the root of the problem that caused us to rollback before.

In summary, it's probably safe to go all the way to the most recent version. In fact, my gut feeling is that the risk is low enough to be outweighed by the benefit of being up-to-date.

**Review by breilly2 — APPROVED** [review]

**Review by cjllanwarne — APPROVED** [review]
Love @breilly2 's analysis. 👍 from me.

**Comment by aednichols** [comment]
Likewise, thank you @breilly2 for the analysis!
